### ISSUE TITLE
[BUG] clang-cl does not support /MP

### ISSUE THREAD

**Issue by zhihaoy** [pybind/pybind11#2835]
## Issue description

```bash
clang-cl: error: argument unused during compilation: '/MP' [-Werror,-Wunused-command-line-argument]
```

I encountered the issue because I added `-Werror`, but pybind11 shouldn't unconditionally add it at the first place. I worked around that issue by setting `pybind11::windows_extras` to empty.

**Comment by henryiii** [comment]
I think this is already being fixed in #2824?

**Comment by zhihaoy** [comment]
I think yes.
